logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/mls/OneOnOneMigrator.kt

```diff
@@ -140,16 +144,26 @@ internal class OneOnOneMigratorImpl(
         ).flatMap { proteusOneOnOneConversations ->
             // We can theoretically have more than one proteus 1-1 conversation with
             // team members since there was no backend safeguards against this
-            proteusOneOnOneConversations.foldToEitherWhileRight(Unit) { proteusOneOnOneConversation, _ ->
-                kaliumLogger.d(
-                    "migrating proteus ${proteusOneOnOneConversation.toLogString()} " +
-                            "to MLS conv ${targetConversation.toLogString()}"
-                )
+            proteusOneOnOneConversations.foldToEitherWhileRight(null as Instant?) { proteusOneOnOneConversation, lastModifiedDate ->
                 messageRepository.moveMessagesToAnotherConversation(
                     originalConversation = proteusOneOnOneConversation,
                     targetConversation = targetConversation
-                )
+                ).map {
+                    // Emit most recent last modified date of the proteus conversations to pass it to the target conversation
+                    conversationRepository.getConversationById(proteusOneOnOneConversation).getOrNull()?.lastModifiedDate.let {
+                        listOfNotNull(lastModifiedDate, it).maxOrNull()
+                    }
+                }
+            }.map { lastModifiedDate ->
+                // Fallback to current time if not found as it means that it's completely new conversation without any history
+                lastModifiedDate ?: currentInstant()
             }
+        }.flatMap { lastModifiedDate ->
+            conversationRepository.updateConversationModifiedDate(targetConversation, lastModifiedDate)
```

Comment:
you can do the same in the query that does the update to avoid updating 
https://github.com/wireapp/kalium/blob/d54cc2881d988b53e362db23fbfe5185a2e0b194/persistence/src/commonMain/db_user/com/wire/kalium/persistence/Messages.sq#L582

Reply:
not sure whether `moveMessages` query should have any side effects and we don't update it for every message type so we would need to duplicate the logic from `persistMessage` and `MessageContent.shouldUpdateConversationOrder()` into sql, also, I think it would be harder to set current time as iso string directly in sql 🤔